resolver/addressable_resolver_test.go

```diff
@@ -360,6 +361,233 @@ func TestGetURI_ObjectReference(t *testing.T) {
+			},
+			dest: duckv1.Destination{
+				Ref: getAddressableRef(),
+				URI: &apis.URL{
+					Path: "/foo",
+				},
+			},
+			wantURI: addressableDNS + "/foo",
+		}, "ref ends with path and no trailing slash and relative URI without leading slash ": {
+			objects: []runtime.Object{
+				getAddressableWithPathAndNoTrailingSlash(),
+			},
+			dest: duckv1.Destination{
+				Ref: getAddressableRef(),
+				URI: &apis.URL{
+					Path: "foo",
+				},
+			},
+			wantURI: addressableDNS + "/foo",
+		}, "ref ends with path and no trailing slash and relative URI with leading slash ": {
+			objects: []runtime.Object{
+				getAddressableWithPathAndNoTrailingSlash(),
+			},
+			dest: duckv1.Destination{
+				Ref: getAddressableRef(),
+				URI: &apis.URL{
+					Path: "/foo",
+				},
+			},
+			wantURI: addressableDNS + "/foo",
+		}, "ref with URI which is absolute URL": {
+			objects: []runtime.Object{
+				getAddressable(),
+			},
+			dest: duckv1.Destination{
+				Ref: getAddressableRef(),
+				URI: &apis.URL{
+					Scheme: "http",
+					Host:   "example.com",
+					Path:   "/foo",
+				},
+			},
+			wantErr: fmt.Errorf("absolute URI is not allowed when Ref or [apiVersion, kind, name] exists"),
+		},
+		"nil url": {
+			objects: []runtime.Object{
+				getAddressableNilURL(),
+			},
+			dest:    duckv1.Destination{Ref: getUnaddressableRef()},
+			wantErr: fmt.Errorf(`url missing in address of %+v`, getUnaddressableRef()),
+		},
+		"nil address": {
+			objects: []runtime.Object{
+				getAddressableNilAddress(),
+			},
+			dest:    duckv1.Destination{Ref: getUnaddressableRef()},
+			wantErr: fmt.Errorf(`address not set for %+v`, getUnaddressableRef()),
+		}, "missing host": {
+			objects: []runtime.Object{
+				getAddressableNoHostURL(),
+			},
+			dest:    duckv1.Destination{Ref: getUnaddressableRef()},
+			wantErr: fmt.Errorf(`hostname missing in address of %+v`, getUnaddressableRef()),
+		}, "missing status": {
+			objects: []runtime.Object{
+				getAddressableNoStatus(),
+			},
+			dest:    duckv1.Destination{Ref: getUnaddressableRef()},
+			wantErr: fmt.Errorf(`address not set for %+v`, getUnaddressableRef()),
+		}, "notFound": {
+			dest:    duckv1.Destination{Ref: getUnaddressableRef()},
+			wantErr: fmt.Errorf(`failed to get ref %+v: %s "%s" not found`, getUnaddressableRef(), unaddressableResource, unaddressableName),
+		}}
+
+	for n, tc := range tests {
+		t.Run(n, func(t *testing.T) {
+			ctx, _ := fakedynamicclient.With(context.Background(), scheme.Scheme, tc.objects...)
+			ctx = addressable.WithDuck(ctx)
+			r := resolver.NewURIResolver(ctx, func(types.NamespacedName) {})
+
+			// Run it twice since this should be idempotent. URI Resolver should
+			// not modify the cache's copy.
+			_, _ = r.URIFromDestinationV1(tc.dest, getAddressable())
+			uri, gotErr := r.URIFromDestinationV1(tc.dest, getAddressable())
+
+			if gotErr != nil {
+				if tc.wantErr != nil {
+					if diff := cmp.Diff(tc.wantErr.Error(), gotErr.Error()); diff != "" {
```

Comment:
`.Error()` is a string, just compare with `got != want`.

Reply:
Sorry, are you saying get rid of cmp.Diff? I like the fact the diff actually makes it easier to understand what the diffs are and not just that they are different. Or maybe I misunderstood what you meant here :)